File: Punto.py
import math

class Punto:
    G = None  

    def __init__(self, x, y):
        self.x = x
        self.y = y

    def coordenada_cartesiana(self):
        print(f"({self.x}, {self.y})")

    def coordenada_polar(self):
        r = math.sqrt(self.x ** 2 + self.y ** 2)
        theta = math.degrees(math.atan2(self.y, self.x))
        print(f"(r={r}, θ={theta}°)")

    def leer_punto(self):
        self.x = float(input())
        self.y = float(input())

    @staticmethod
    def main():
        p = Punto(3, 4)
        p.coordenada_cartesiana()
        p.coordenada_polar()
        p.leer_punto()
        p.coordenada_cartesiana()
        p.coordenada_polar()


Punto.G = Punto(0, 0)

Punto.main()

File: Punto.java
import java.util.Scanner;

class Punto {
    private double x;
    private double y;
    private static final Punto G = new Punto(0, 0);

    Punto(double x, double y) {
        this.x = x;
        this.y = y;
    }

    void coordenadaCartesiana() {
        System.out.println("(" + x + ", " + y + ")");
    }

    void coordenadaPolar() {
        double r = Math.sqrt(x * x + y * y);
        double theta = Math.atan2(y, x);
        System.out.println("(r=" + r + ", θ=" + Math.toDegrees(theta) + "°)");
    }

    void leerPunto() {
        Scanner scanner = new Scanner(System.in);
        x = scanner.nextDouble();
        y = scanner.nextDouble();
    }

    public static void main(String[] args) {
        Punto p = new Punto(3, 4);
        p.coordenadaCartesiana();
        p.coordenadaPolar();
    }
}
